### PR TITLE
Upgrade wrensec-ui to version 23 in UI modules.

### PR DESCRIPTION
**Pull request by krausvo1**
I have upgraded `wrensec-ui` dependency to the latest release `23.0.1` in `openam-ui/openam-ui-api` and `openam-ui/openam-ui-ria` modules.

This PR primarily focuses on switching from Grunt to Gulp and adjusting code to `wrensec-ui` changes and dependency upgrades (e.g. lodash).

I have also upgraded CodeMirror to version 6. This could perhaps be implemented better so the usage is more convenient, but I don't think that's crucial for now since the editor is used in a single view only. Former ForgeRock editor theme is no longer used since it was deleted from `wrensec-ui` and the default CodeMirror theme has improved a little since version 4.

I have tested admin UI thoroughly, everything seems to work fine after the changes.

### CODE REVIEW

File: openam-ui/openam-ui-ria/src/main/js/org/forgerock/openam/ui/common/components/TabComponent.js [hunk 1]
```diff
@@ -47,10 +47,10 @@ define([
...
                 throw new TypeError("[TabComponent] \"tabs\" argument is an empty Array.");
             }
-            _(options.tabs)
-                .each(_.partial(has, "id"))
-                .each(_.partial(has, "title"))
-                .value();
+            _(options.tabs).each(function (tab) {
+                has(tab, "id");
+                has(tab, "title");
+            });
```
Comment (krausvo1) [inline openam-ui/openam-ui-ria/src/main/js/org/forgerock/openam/ui/common/components/TabComponent.js:54]: `each` immediately runs the sequence chain, so the two `each` methods here cannot be chained anymore.

**Comment by krausvo1** [comment]
I have squashed few commits. The PR is now ready for review.

**Comment by krausvo1** [comment]
Loading indicator spinner sometimes stays displayed when it should be hidden after the upgrade. This is upstream issue (see https://github.com/WrenSecurity/wrensec-ui/pull/36) that I think can be solved separately from this PR.

**Review by pavelhoral — CHANGES_REQUESTED** [review]

File: openam-ui/openam-ui-ria/src/test/js/org/forgerock/openam/ui/common/util/PromiseTest.js [hunk 2]
```diff
@@ -48,97 +57,120 @@ define([
...
                     });
-                    it("doesn't resolve the returned promise if only the second promise is resolved", () => {
+                    // eslint-disable-next-line max-len
+                    it("doesn't resolve the returned promise if only the second promise is resolved", async function () {
                         d2.resolve();
+
+                        await d2.promise();
+                        // Make sure spies are not checked too early
+                        this.timeout(2000);
+                        await waitFor($, 1500);
```
Comment (pavelhoral) [inline openam-ui/openam-ui-ria/src/test/js/org/forgerock/openam/ui/common/util/PromiseTest.js:137]: This is very strange construct. Why do we need to wait? Also the "await all" construct seems strange as well. It should be possible to simply await the promise we want to be resolved, or am I missing something?

File: openam-ui/openam-ui-ria/src/main/resources/templates/common/components/tab/TabComponentTemplate.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
     <ul class="nav nav-tabs" role="tablist">
         {{#each tabs}}
             <li>
-                <a href="#" data-tab-id="{{this.id}}" role="tab" data-toggle="tab" aria-expanded="false">{{this.title}}</a>
+                <a href="#{{this.id}}" data-tab-id="{{this.id}}" role="tab" data-toggle="tab" aria-expanded="false">{{this.title}}</a>
```
Comment (pavelhoral) [inline openam-ui/openam-ui-ria/src/main/resources/templates/common/components/tab/TabComponentTemplate.html:5]: Is this change necessary?

Comment (krausvo1) [reply openam-ui/openam-ui-ria/src/main/resources/templates/common/components/tab/TabComponentTemplate.html:5]: It is, see https://jquery.com/upgrade-guide/3.0/#breaking-change-jquery-quot-quot-and-find-quot-quot-are-invalid-syntax. It would throw an error on tab click.

File: openam-ui/openam-ui-ria/package.json [hunk 4]
```diff
@@ -19,22 +19,11 @@
     "@babel/eslint-parser": "^7.22.15",
```
[the remark this answers is not in the document]
Comment (krausvo1) [reply openam-ui/openam-ui-ria/package.json:19]: Noted

File: openam-ui/openam-ui-ria/gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    dest: join(TARGET_PATH, "org/forgerock/openam/ui/admin/utils/CodeMirror.js")
+}));
+
+/**
+ * Include the version of AM in the index file.
+ *
+ * This is needed to force the browser to refetch JavaScript files when a new version of AM is deployed.
+ */
+gulp.task("build:version", async () => {
+    await gulp.src(`${TARGET_PATH}/index.html`)
```
[the remark this answers is not in the document]
Comment (krausvo1) [reply openam-ui/openam-ui-ria/gulpfile.js:132]: You're right, I'll fix it.

File: openam-ui/openam-ui-api/pom.xml [hunk 6]
```diff
@@ -45,13 +45,46 @@
         </resources>
 
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+
+                <executions>
+                    <execution>
+                        <id>unpack</id>
```
[the remark this answers is not in the document]
Comment (krausvo1) [reply openam-ui/openam-ui-api/pom.xml:54]: You are right, I'm not really sure why I added it here in the first place.

[the remark this answers is not in the document]
Comment (krausvo1) [reply openam-ui/openam-ui-api/pom.xml]: Removed.

File: openam-ui/openam-ui-ria/src/main/js/org/forgerock/openam/ui/admin/views/realms/sessions/SessionsTable.jsx [hunk 7]
```diff
@@ -13,6 +13,7 @@
  *
  * Copyright 2016 ForgeRock AS.
  * Portions copyright 2024 Wren Security.
+ * Portions copyright 2024 Wren Security.
```
[the remark this answers is not in the document]
Comment (krausvo1) [reply openam-ui/openam-ui-ria/src/main/js/org/forgerock/openam/ui/admin/views/realms/sessions/SessionsTable.jsx]: Removed

File: openam-ui/pom.xml [hunk 8]
```diff
@@ -27,7 +27,7 @@
     <artifactId>openam-ui</artifactId>
     <packaging>pom</packaging>
 
-    <name>Wren:AM - UI Parent</name>
+    <name>Wren:AM - UI - Parent</name>
```
[the remark this answers is not in the document]
Comment (krausvo1) [reply openam-ui/pom.xml:30]: Reverted

File: openam-ui/openam-ui-ria/gulpfile.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (krausvo1) [reply openam-ui/openam-ui-ria/gulpfile.js]: Fixed

File: openam-ui/openam-ui-ria/src/test/js/org/forgerock/openam/ui/common/util/PromiseTest.js (hunk 2 above)
Comment (krausvo1) [reply openam-ui/openam-ui-ria/src/test/js/org/forgerock/openam/ui/common/util/PromiseTest.js]: Rewritten the "waiting" part with setTimeout.

The "all" variable is the promise returned by `Promise.all`, that's the promise we want to test here I believe. Therefore we need wait till it's resolved and then we can call the assert functions.

**Review by pavelhoral — APPROVED** [review]
LGTM
